psi_cardinality/go/server/server.go

```diff
@@ -0,0 +1,125 @@
+package server
+
+/*
+#cgo LDFLAGS: -L${SRCDIR}/../extern/lib/
+#cgo LDFLAGS: -Wl,-rpath,${SRCDIR}/../extern/lib
+#cgo CFLAGS: -I${SRCDIR}/../../cpp -I${SRCDIR}/../../cpp/util
+#cgo LDFLAGS:  -lpsi_cardinality_server -lec_commutative_cipher -lec_util -lelgamal -lbloom_filter -lbn_util -lcrypto -lopenssl_init -lstatus -lglog -lgflags -lstrings -linternal -lraw_logging_internal -lthrow_delegate -lint128
+
+#include "psi_cardinality_server_c.h"
+*/
+import "C"
+import (
+	"errors"
+)
+
+//PSICardinalityServer context
+type PSICardinalityServer struct {
+	context C.psi_cardinality_server_ctx
+}
+
+//CreateWithNewKey returns a new PSI server
+func CreateWithNewKey() (*PSICardinalityServer, error) {
+	psiServer := new(PSICardinalityServer)
+	psiServer.context = C.psi_cardinality_server_create_with_new_key()
+	if psiServer.context == nil {
+		return nil, errors.New("failed to create server context")
```

Comment:
If we get the error message from C as suggested above then we could forward it here.